core-java-security/src/main/java/com/baeldung/ssl/EnableTLSv12.java

```
@@ -0,0 +1,134 @@
+        port = Integer.parseInt(portNumber);
+    }
+
+    private void handleCommunication(SSLSocket socket, String usedTLSProcess) throws IOException {
+        System.out.print("Enabled TLS v1.2 on " + usedTLSProcess);
+        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(socket.getOutputStream())));
+        out.println("GET / HTTP/1.0");
+        out.println();
+        out.flush();
+        if (out.checkError())
+            System.out.println("SSLSocketClient:  java.io.PrintWriter error");
+
+        BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+
+        String inputLine;
+        while ((inputLine = in.readLine()) != null)
+            System.out.println(inputLine);
+
+        in.close();
+        out.close();
+        socket.close();
+    }
+
+    public void enableTLSv12UsingSSLParameters() {
+        try {
+            SSLSocketFactory socketFac = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            SSLSocket sslSocket = (SSLSocket) socketFac.createSocket(url.trim(), port);
+            SSLParameters params = new SSLParameters();
+            params.setProtocols(new String[] { "TLSv1.2" });
+            sslSocket.setSSLParameters(params);
+            sslSocket.startHandshake();
+            handleCommunication(sslSocket, "SSLSocketFactory-SSLParameters");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void enableTLSv12UsingProtocol() throws IOException {
+        try {
+            SSLSocketFactory socketFac = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            SSLSocket sslSocket = (SSLSocket) socketFac.createSocket(url.trim(), port);
+            sslSocket.setEnabledProtocols(new String[] { "TLSv1.2" });
+            sslSocket.startHandshake();
+            handleCommunication(sslSocket, "SSLSocketFactory-EnabledProtocols");
+        } catch (Exception e) {
+            e.printStackTrace();
```

Let's either throw the exception the signature or wrap this exception in something like an `IllegalStateException`. Same for the other places where you have an `e.printStackTrace`